src/core/ext/filters/client_channel/http_proxy.cc

```diff
@@ -107,6 +107,17 @@ char* GetHttpProxyServer(const grpc_channel_args* args, char** user_cred) {
   return proxy_name;
 }
 
+// Adds the default port (443) if target does not contain a port.
+std::string MaybeAddDefaultPort(absl::string_view target) {
+  absl::string_view host;
+  absl::string_view port;
+  SplitHostPort(target, &host, &port);
+  if (port.empty()) {
+    return JoinHostPort(host, 443);
```

Comment:
I'd ideally prefer not to hard-code the default this way, independently of the default in the DNS resolver that specifies the same thing.  But unfortunately, I don't have a better suggestion for how to handle this, so I guess we can live with it.

Reply:
I can make it a constant. Something like kDefaultSecurePort that is shared between the resolver and this.

Reply:
That sounds fine.  It's already defined independently in both of the DNS resolver implementations:

https://github.com/grpc/grpc/blob/c97da0fc25b9bd8d1eead7b76d7a8a1b60625956/src/core/ext/filters/client_channel/resolver/dns/native/dns_resolver.cc#L51

https://github.com/grpc/grpc/blob/c97da0fc25b9bd8d1eead7b76d7a8a1b60625956/src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc#L63